File: my_recognition_python/my_recognition.py
#!/usr/bin/python3
import jetson.inference
import jetson.utils

import argparse

parser = argparse.ArgumentParser()
parser.add_argument("filename", type=str, help="filename of the image to process")
parser.add_argument("--network", type=str, default="googlenet", help="model to use, can be: googlenet, resnet-18, ect. (see --help for others)")
opt = parser.parse_args()

# Loading the image from Disk
# loadImageRGBA()関数を使うことでGPUメモリにイメージ（画像）をロードできる
# ロードされたイメージはCPUとGPUの両方にマップされている共有メモリに保存される
img, width, height = jetson.utils.loadImageRGBA(opt.filename)

# load the recognition network 画像認識ネットワークの読み込み
# imageNetオブジェクトを使ってTensorRTで分類モデルを読み込む
net = jetson.inference.imageNet(opt.network)

# 画像の分類
# 画像本体とサイズを渡し、TensorRTで推論を実行する
# 戻り値：画像が認識されたオブジェクトクラスの整数インデックスを含むタプルと信頼値
class_idx, confidence = net.Classify(img, width, height)

# 結果の解釈
# クラスの説明を取得して、分類の結果を出力する
class_desc = net.GetClassDesc(class_idx)
# print out the result
print("image is recognized as '{:s}' (class #{:d}) with {:f}% confidence".format(class_desc, class_idx, confidence * 100))

File: my_recognition_cpp/my_recognition.cpp
#include <jetson-inference/imageNet.h>
#include <jetson-utils/loadImage.h>

int main(int argc, char** argv)
{
	if (argc < 2)
	{
		printf("my-recognition: expected image filename as argument\n");
		printf("example usage : ./my_recognition my_image.jpg\n");
		return 0;
	}

	const char* imgFilename = argv[1];
	
	// Loading the Image from Disk
	float* imgCPU = NULL;	// CPU pointer to floating-point RGBA image data
	float* imgCUDA = NULL;	// GPU pointer to floating-point RGBA image data
	int imgWidth = 0;		// width of the image (in pixels)
	int imgHeight = 0;		// height of the image (in pixels)

	// load the image from disk as float4 RGBA (32 bits per channel, 128 bits per pixel)
	if (!loadImageRGBA(imgFilename, (float4**)&imgCPU, (float4**)&imgCUDA, &imgWidth, &imgHeight))
	{
		printf("failed to load image '%s'\n", imgFilename);
		return 0;
	}
	
	// Loading the Image Recognition Network
	imageNet* net = imageNet::Create(imageNet::GOOGLENET);
	if (!net)
	{
		printf("failed to load image recognition network\n");
		return 0;
	}

	// Classifying the Image
	float confidence = 0.0;
	const int classIndex = net->Classify(imgCUDA, imgWidth, imgHeight, &confidence);

	// Intepreting the Results
	if (classIndex >= 0)
	{
		const char* classDescription = net->GetClassDesc(classIndex);

		printf("image is recognized as '%s' (class #%i) with %f%% confidence\n", classDescription, classIndex, confidence * 100.0f);
	}
	else
	{
		printf("failed to classify image\n");
	}

	delete net;

	return 0;
}

